x-pack/plugins/lens/public/indexpattern_datasource/datapanel.tsx

```diff
@@ -488,23 +561,83 @@ export const InnerIndexPatternDataPanel = function InnerIndexPatternDataPanel({
         responsive={false}
       >
         <EuiFlexItem grow={false}>
-          <div className="lnsInnerIndexPatternDataPanel__header">
-            <ChangeIndexPattern
-              data-test-subj="indexPattern-switcher"
-              trigger={{
-                label: currentIndexPattern.title,
-                title: currentIndexPattern.title,
-                'data-test-subj': 'indexPattern-switch-link',
-                fontWeight: 'bold',
-              }}
-              indexPatternId={currentIndexPatternId}
-              indexPatternRefs={indexPatternRefs}
-              onChangeIndexPattern={(newId: string) => {
-                onChangeIndexPattern(newId);
-                clearLocalState();
-              }}
-            />
-          </div>
+          <EuiFlexGroup
+            gutterSize="s"
+            alignItems="center"
+            className="lnsInnerIndexPatternDataPanel__header"
+          >
+            <EuiFlexItem grow={true}>
```

Comment:
Currently, there is a bug where the index pattern selector isn't truncating to ellipsis as it should. So if the width of the selected index pattern name is long, it pushes the new context menu button out of the sidebar, thus hiding it from users.

![image](https://user-images.githubusercontent.com/3884767/110865899-55211680-8292-11eb-930a-ab8b15f050d8.png)

This can be corrected by applying a style of `min-width: 0;` to this particular `EuiFlexItem`.